### PR TITLE
Dictionaries implement copyOrUpdate

### PR DESCRIPTION
**Pull request by clementetb**
With this PR we add support for copyToRealmOrUpdate, a method needed to copy RealmObjects with a primary key.

### CODE REVIEW

**Review by edualonso — APPROVED** [review]
Looks great. Thanks a lot for the help! 🙏

File: realm/realm-library/src/androidTest/kotlin/io/realm/ManagedDictionaryTesters.kt [hunk 1]
```diff
@@ -364,6 +366,28 @@ class ManagedDictionaryTester<T : Any>(
         }
     }
 
+    override fun copyToRealmOrUpdate() {
```
Comment (rorbech) [inline realm/realm-library/src/androidTest/kotlin/io/realm/ManagedDictionaryTesters.kt:369]: This only tests the copy part, not the update part.

**Review by rorbech — CHANGES_REQUESTED** [review]
There is no tests for the update part of `copyOrUpdate`, or am I missing something?

File: realm/realm-library/src/androidTest/kotlin/io/realm/ManagedDictionaryTesters.kt (hunk 1 above)
Comment (edualonso) [reply realm/realm-library/src/androidTest/kotlin/io/realm/ManagedDictionaryTesters.kt:369]: That's right 😱 You should just grab hold of the object you just created, modify the dictionary field, call `copyToRealmOrUpdate` again, and assert again. Sorry I missed this one :(

Comment (clementetb) [reply realm/realm-library/src/androidTest/kotlin/io/realm/ManagedDictionaryTesters.kt:369]: I did miss it too

**Review by rorbech — COMMENTED** [review]
@clementetb I think it is a bit convoluted (at least I have hard time reasoning about it). Why not just follow Eduardo's proposal of modifying the `manualInstance` after you have verified that it is copied correctly after the first `copyToRealmOrUpdate` and then reverified that the updates to `manualInstance` is in fact updated (on the same instance) after a second `copyToRealmOrUpdate`. 

```
val manulalInstance = ...
val managedInstance = copyOrUpdate ...
// verify managedInstance 

// update manual instance
manualInstance....
// verify managed instance against updates
```

**Comment by clementetb** [comment]
I was initializing it with an empty object (the PrimaryKeyDictionaryContainer has a default PK). But that's true that I wasn't validating that the data was actually stored in the DB. Moved to what Edu suggested👌

**Pull request merged**